apps/main/src/modules/hall/hall.service.ts

```diff
@@ -45,26 +121,41 @@ export class HallService {
         return hallPlan;
     }
 
-    async mapHallPlan(hallPlan: HallPlan, sessionId: string): Promise<HallPlanResponseDto> {
+    async mapHallPlan(hallPlan: HallPlan, sessionId?: string): Promise<HallPlanResponseDto> {
         const mappedHallPlan: HallPlanResponseDto = {};
 
         for (const key in hallPlan) {
             mappedHallPlan[key] = await Promise.all(
                 hallPlan[key].map(async (seat, idx) => {
-                    const hallSeat = await this.hallSeatService.findOne({ id: seat.id }, ['sessionHallSeats', 'sessionHallSeats.session']);
-                    const sessionHallSeat = hallSeat.sessionHallSeats.find((sessionHallSeat) => sessionHallSeat.session.id === sessionId);
-                    let reserved = false;   
-                    const reservedSeat = await this.reservationHallSeatService.findOne({
-                        hallSeat: { id: hallSeat.id },
-                        session: { id: sessionId }
-                    });
-                    if (reservedSeat) {
-                        reserved = true;
+                    const hallSeat = await this.hallSeatService.findOne({ id: seat.id }, [
+                        'sessionHallSeats',
+                        'sessionHallSeats.session'
+                    ]);
+
+                    // Find the sessionHallSeat only if sessionId is provided
+                    let sessionHallSeat = null;
+                    if (sessionId) {
```

Comment:
I believe you can combine this if (sessioId) statement with statement on the line 145